tests/unit/test_host_select.py

```diff
@@ -200,3 +202,18 @@ def test_condemned_host_ambiguous(mock_glbl_cfg):
             '''
         )
     assert 'ambiguous host' in excinfo.value.msg
+
+
+def test_get_metrics_no_hosts_error(caplog):
+    """It should handle SSH errors.
+
+    If a host is not contactable then it should be shipped.
+    """
+    caplog.set_level(logging.WARN, 'cylc')
```

Comment:
Minor niggle, to avoid hard-coded string:
```
from cylc.flow import CYLC_LOG
```

```suggestion
    caplog.set_level(logging.WARN, CYLC_LOG)
```